server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -431,6 +451,25 @@ public void waitForTaskCompletion(Task task, long untilInNanos) {
             this.task = task;
         }
 
+        public void registerChildNode(DiscoveryNode node) {
```

Comment:
I wonder if we also need a method to unregister nodes after calls are completed (would need a counter per node), so that a request that successively reaches out  to a lot of nodes does not need to send ban requests to many of these nodes where the  requests have been completed).